### ISSUE TITLE
Insert query with arguments inside a plpgsql function fails to evaluate parameters

### ISSUE THREAD

**Issue by saicitus** [citusdata/citus#752]
Version: Citus 5.2
**Steps to reproduce:**
```
create table table1 (
key int not null,
value int
);
select master_create_distributed_table('table1','key','hash');
select master_create_worker_shards('table1',16,1);

create table table2 (
key int not null,
value int
);
select master_create_distributed_table('table2','key','hash');
select master_create_worker_shards('table2',16,1);

CREATE OR REPLACE FUNCTION insert_func( key_arg int,
val_arg int
) RETURNS void as $$
DECLARE
distribution_col CONSTANT INTEGER := key_arg;
BEGIN
INSERT INTO table1 (key, value)
VALUES (distribution_col,val_arg);
END;
$$ language plpgsql;
select insert_func(1,3);
```
**Error:**
```
WARNING:  could not determine data type of parameter $1
CONTEXT:  while executing command on <ip>:5432
SQL statement "INSERT INTO table1 (key, value)
VALUES (distribution_col,val_arg)"
PL/pgSQL function insert_func(integer,integer) line 5 at SQL statement
ERROR:  could not modify any active placements
CONTEXT:  SQL statement "INSERT INTO table1 (key, value)
VALUES (distribution_col,val_arg)"
PL/pgSQL function insert_func(integer,integer) line 5 at SQL statement
```

Even if I hardcode the partition column as below, similar error persists:
```
CREATE OR REPLACE FUNCTION insert_func( key_arg int,
val_arg int
) RETURNS void as $$
DECLARE
distribution_col CONSTANT INTEGER := key_arg;
BEGIN
INSERT INTO table1 (key, value)
VALUES (1,val_arg);
END;
$$ language plpgsql;
```

A simple work around is to wrap the INSERT statement in an EXECUTE.
```
CREATE OR REPLACE FUNCTION test_schema.insert_func( key_arg int,
val_arg int
) RETURNS void as $$
DECLARE
distribution_id CONSTANT INTEGER := key_arg;
BEGIN
EXECUTE format('INSERT INTO test_schema.table1 (key, value)
VALUES (%s,%s)',distribution_id,val_arg);
END;
$$ language plpgsql;
```

**Comment by saicitus** [comment]
Came up when I was testing this closed issue #700

**Comment by anarazel** [comment]
@metdos ^
